### ISSUE TITLE
Fix S2996 FN: Support generic attributes

### ISSUE THREAD

**Issue by gregory-paidis-sonarsource** [SonarSource/sonar-dotnet#6395]
### Description

FN for S2996 (ThreadStaticWithInitializer).

Using C#11 generic attributes, the rule is not raising an issue.

### Repro steps

```csharp
    public class ThreadStaticWithInitializer
    {
        public class Foo
        {
            [ThreadStaticAttribute<int>]
            public static object PerThreadObject = new object(); // FN
        }

        public class ThreadStaticAttribute<T> : ThreadStaticAttribute { }
    }
```

### Expected behavior

The rule should raise an issue.

### Actual behavior

The rule does not raise an issue.
